### ISSUE TITLE
Support for video profiles to enable lower resolution video capture

### ISSUE THREAD

**Issue by djee-ms** [webrtc-uwp/webrtc-uwp-sdk#170]
Some devices like HoloLens 2 only report a subset of the supported resolutions when opening the video capture device with the default video profile. In the case of HoloLens 2 this prevents the developers from accessing lower resolution video feeds, which has an impact on performance and also disables Mixed Reality Capture (which is available only if <= 1080p).

The [initialization code in `webrtc\xplatform\webrtc-apis\windows\wrapper\impl_webrtc_VideoCapturer.cpp`](https://github.com/webrtc-uwp/webrtc-apis/blob/0d303d62f7e6483cf4b57c24ef295c487b0c28dd/windows/wrapper/impl_webrtc_VideoCapturer.cpp#L649) does not specify any value for the [`MediaCaptureInitializationSettings.VideoProfile`](https://docs.microsoft.com/en-us/uwp/api/windows.media.capture.mediacaptureinitializationsettings.videoprofile) property when calling [`InitializeAsync(settings)`](https://docs.microsoft.com/en-us/uwp/api/windows.media.capture.mediacapture.initializeasync#Windows_Media_Capture_MediaCapture_InitializeAsync_Windows_Media_Capture_MediaCaptureInitializationSettings_).

**Comment by kspark-scott** [comment]
This issue is a showstopper for us in porting our RemoteSpark application (https://www.kognitivspark.com/remotespark/) to HoloLens 2.  Our application depends entirely on Mixed Reality Capture, and the high resolution video capture consumes processing resources that we really need for rendering and other things.

My early HL2 experiments included a few stand-alone sample apps in which I found that the low power 1280x720 mode (and others) become available when you set `SharingMode` to `MediaCaptureSharingMode::SharedReadOnly` in the media capture initialization settings.  However, my attempts to make the same adjustment in `impl_webrtc_VideoCapturer.cpp` resulted in a sharing violation error when setting video stream properties, so perhaps that's not the way to go.  I am really not familiar with this WebRTC code so I'm hoping there is a straightforward solution waiting there for someone more knowledgeable.

**Comment by djee-ms** [comment]
@kspark-scott : I didn't try sharing, but by tweaking the code in the video capturer I was able to select a video profile and get other resolutions and framerates locally. This just need to be properly exposed in the API. I _think_ accessing a lower mode by setting the share mode is just a side-effect, and video profile selection is the proper way to implement that feature. That being said, sharing mode is also a feature you are looking for in addition as I understand, so we should look into this too.

**Comment by kspark-scott** [comment]
@djee-ms Yes, I suggested setting the share mode only because it appeared to produce the desired result at the end of my fiddle-and-tweak session.  If video profile selection looks to be the proper path to then that sounds good to me too.
